analysis/src/batcher.rs

```diff
@@ -1,19 +1,22 @@
 use std::marker::PhantomData;
 
+use ast::analysis::{
+    AnalysisASMFile, BatchMetadata, Incompatible, IncompatibleSet, ProgramStatement,
+};
 use itertools::Itertools;
 use number::FieldElement;
-use parser::asm_ast::{
-    batched::{ASMStatementBatch, BatchedASMFile, Incompatible, IncompatibleSet},
-    ASMFile, ASMStatement,
-};
+
+pub fn batch<T: FieldElement>(file: AnalysisASMFile<T>) -> AnalysisASMFile<T> {
+    ProgramBatcher::default().batch(file)
+}
 
 #[derive(Default)]
```

Comment:
I refactored the batcher to work with the new analysis structure: now batches as stored separately from statements